blocks/extensions/monitoring.py

```diff
@@ -139,3 +140,17 @@ def do(self, callback_name, *args):
             self.add_records(self.main_loop.log,
                              self._buffer.get_aggregated_values().items())
             self._buffer.initialize_aggregators()
+
+
+class CrossEntropy(MonitoredQuantity):
+    def __init__(self, **kwargs):
+        super(CrossEntropy, self).__init__(**kwargs)
+        self.total_cross_entropy, self.examples_seen = 0.0, 0
+
```

Comment:
When trying to play with your code I realized that a `reset` (or `initialize` step is missing). Every time before monitoring the value of `MonitoredQuantity` should be somehow initialized.